node/building_blocks.re

```diff
@@ -96,18 +119,62 @@ let is_signable = (state, block) => {
   && is_current_producer(state, ~key_hash=block.author)
   && !has_next_block_to_apply(state, ~hash=block.hash)
   && all_main_ops_are_known
-  && contains_only_trusted_add_validator_op(block.side_chain_ops);
+  && contains_only_trusted_add_validator_op(block.side_chain_ops)
+  && block_has_signable_state_root_hash(~current_time, state, block);
 };
 
 let sign = (~key, block) => Block.sign(~key, block);
 
-let produce_block = state =>
+/** The state_root_min_timeout determines how often the block producer
+    starts hashing a new state root. It is closely related to the
```

Comment:
This comment is kind of weird, on the async version it doesn't determines, but here it does, but it does determines when it hashes, not when it starts hashing. Slightly different.